pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -976,17 +987,11 @@ func (gce *GCECloud) firewallObject(name, region, desc string, sourceRanges nets
 	return firewall, nil
 }
 
-// If the node tags to be used for this cluster have been predefined in the
-// provider config, just use them. Otherwise, grab all tags from all relevant
-// instances:
+// We grab all tags from all instances being added to the pool.
 // * The longest tag that is a prefix of the instance name is used
 // * If any instance has a prefix tag, all instances must
 // * If no instances have a prefix tag, no tags are used
 func (gce *GCECloud) computeHostTags(hosts []*gceInstance) ([]string, error) {
-	if len(gce.nodeTags) > 0 {
```

Comment:
please add a comment about this method like: "invoking this method to get host tags is risky since it depends on the format of the host names in the cluster. Only use it as a fallback if gce.nodeTags is unspecified."